File: backend/src/main/java/com/bookflow/report/service/ReportService.java
package com.bookflow.report.service;

import com.bookflow.report.dto.response.DailyReportResponse;
import com.bookflow.report.dto.response.MonthlyReportResponse;

import java.time.LocalDate;

public interface ReportService {

    DailyReportResponse getDailyReport(
        Long companyId,
        LocalDate date
    );

    MonthlyReportResponse getMonthlyReport(
        Long companyId,
        int year,
        int month
    );
}

File: frontend/src/services/reportService.js
import api from './api'

export const reportService = {
  getDaily: (companyId, date) =>
    api.get(`/api/v1/companies/${companyId}/reports/daily?date=${date}`),
  
  getMonthly: (companyId, year, month) =>
    api.get(`/api/v1/companies/${companyId}/reports/monthly?year=${year}&month=${month}`),
}